src/nvim/api/win_config.c

```diff
@@ -330,7 +333,80 @@ static bool parse_float_bufpos(Array bufpos, lpos_T *out)
   return true;
 }
 
-static void parse_border_style(Object style, FloatConfig *fconfig, Error *err)
+static void parse_border_title(Object title, Object title_pos, Object width, FloatConfig *fconfig, Error *err)
+{
+  if (!parse_title_pos(title_pos, fconfig, err)) {
+    return;
+  }
+
+  if (title.type == kObjectTypeString) {
+    fconfig->title_text = xstrdup(title.data.string.data);
+    int hl_id = object_to_hl_id(title, "border title highlight", err);
+    fconfig->title_hi_id = hl_id;
+    fconfig->title = true;
+    return;
+  }
+
+  if (title.type != kObjectTypeArray) {
+    api_set_error(err, kErrorTypeValidation, "title must be string or dictionary");
+    return;
+  }
+
+  if (title.type == kObjectTypeArray && title.data.array.size == 0) {
+    api_set_error(err, kErrorTypeValidation, "title must not be an empty dictionary");
```

Comment:
You're checking the type above, `title.type` is always `kObjectTypeArray` here.
```suggestion
  if (title.data.array.size == 0) {
    api_set_error(err, kErrorTypeValidation, "title cannot be an empty array");
```